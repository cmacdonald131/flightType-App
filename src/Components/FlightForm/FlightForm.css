* {
  box-sizing: border-box;
}

.aircraft {
  margin-top: 20px;
  margin-bottom: 40px;
  text-align: center;
}

.AddFlight {
  background: linear-gradient(rgba(0, 0, 0, 0.3), rgba(0, 0, 0, 0.3)), url(../../Images/wing.jpg);
  background-position: center center;
  background-size: cover;
  color: #ffffff;
  height: 100vh;
}

button {
    margin-right: 10px;
}

@media screen and (min-width: 700px) {
  .AddFlight {
    background: linear-gradient(rgba(0, 0, 0, 0.3), rgba(0, 0, 0, 0.3)), url(../../Images/flight.jpg);
    background-position: center center;
    background-size: cover;
    color: #ffffff;
    height: 100vh;
  }

  button {
      margin-right: 20px;
  }

  .aircraft {
      width: 60%;
      margin-right: auto;
      margin-left: auto;
  }
}
