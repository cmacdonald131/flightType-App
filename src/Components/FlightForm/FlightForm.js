import React, { Component } from 'react'
import config from '../../config'
import uuid from 'uuid/v4'
import TokenService from '../../Services/token-service'
import './FlightForm.css'
import Navbar from '../Navbar/Navbar'

export default class FlightForm extends Component {
    //creates new flight from information input and posts it to flights endpoint
    onSubmit = (e) => {
        e.preventDefault();
        const name = e.target.name.value;
        const departure_airport = e.target.departure_airport.value;
        const arrival_airport = e.target.arrival_airport.value;
        const company = e.target.company.value;
        const arrival_time = e.target.arrival_time.value;
        const departure_time = e.target.departure_time.value;
        const id = uuid;

        fetch(`${config.API_ENDPOINT}/flights`, {
            method: "Post",
            body: JSON.stringify({
                name: name,
                departure_airport: departure_airport,
                arrival_airport: arrival_airport,
                company: company,
                arrival_time: arrival_time,
                departure_time: departure_time,
                id: id,

            }),
            headers: {
                'Authorization': `Bearer ${TokenService.getAuthToken()}`,
                'content-type': 'application/json'
            }
        })

            .then(flight => {
                return flight.json()
            }).then(data => {
                this.props.history.push('/user')
            })

    }

    onCancel = (e) => {
        e.preventDefault();
        this.props.history.push('/user')
    }

    render() {
        return (
            <div className="AddFlight">
                <Navbar />
                <form onSubmit={this.onSubmit} className="add-flight">
                    <section className="aircraft">
                        <h1 style={{ textAlign: "center", fontSize: "30px" }}>Create your flight</h1>
                        <label htmlFor="FlightForm" style={{ textAlign: "left" }}>Flight Name</label>
                        <input
                            className='search-box'
                            name='name'
                            type='text'
                            placeholder='Input flight name'
                            required
                            id='FlightForm'
                        >
                        </input>
                        <label htmlFor="FlightForm" style={{ textAlign: "left" }}>Departure Airport</label>
                        <input
                            className='search-box'
                            name='departure_airport'
                            type='text'
                            placeholder='Input departure airport'
                            required
                            id='FlightForm'
                        >
                        </input>
                        <label htmlFor="FlightForm" style={{ textAlign: "left" }}>Arrival Airport</label>
                        <input
                            className='search-box'
                            name='arrival_airport'
                            type='text'
                            placeholder='Input arrival airport'
                            required
                            id='FlightForm'
                        >
                        </input>
                        <label htmlFor="FlightForm" style={{ textAlign: "left" }}>Airline</label>
                        <input
                            className='search-box'
                            name='company'
                            type='text'
                            placeholder='Input airline company'
                            required
                            id='FlightForm'
                        >
                        </input>
                        <label htmlFor="FlightForm" style={{ textAlign: "left" }}>Arrival Time</label>
                        <input
                            className='search-box'
                            name='arrival_time'
                            type='time'
                            placeholder='Input arrival time'
                            required
                            id='FlightForm'
                        >
                        </input>
                        <label htmlFor="FlightForm" style={{ textAlign: "left" }}>Departure Time</label>
                        <input
                            className='search-box'
                            name='departure_time'
                            type='time'
                            placeholder='Input departure time'
                            required
                            id='FlightForm'
                        >
                        </input>
                        <button >Create Flight</button>
                        <button type="reset" className="reset">Reset</button>
                        <button onClick={this.onCancel} type="cancel" className="cancel">Cancel</button>
                    </section>
                </form>
            </div>
        )
    }
}