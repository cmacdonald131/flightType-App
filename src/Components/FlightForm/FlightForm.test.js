import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import FlightForm from './FlightForm'


describe(`FlightForm component`, () => {
  const props = {
    name: 'test-name',
    departure_airport: 'test-departure_airport',
    arrival_airport: 'test-arrival_airport',
    departure_time: 'test-departure_time',
    arrival_time: 'test-arrival_time',
    company: 'test-company',
    'data-other': 'test-other-prop'
  }

  it('renders a form.FlightForm by default', () => {
    const wrapper = shallow(<FlightForm />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the FlightForm given props', () => {
    const wrapper = shallow(<FlightForm {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})