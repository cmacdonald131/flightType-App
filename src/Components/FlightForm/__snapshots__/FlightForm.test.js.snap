// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlightForm component renders a form.FlightForm by default 1`] = `
<div
  className="AddFlight"
>
  <Navbar />
  <form
    className="add-flight"
    onSubmit={[Function]}
  >
    <section
      className="aircraft"
    >
      <h1
        style={
          Object {
            "fontSize": "30px",
            "textAlign": "center",
          }
        }
      >
        Create your flight
      </h1>
      <label
        htmlFor="FlightName"
        style={
          Object {
            "textAlign": "left",
          }
        }
      >
        Flight Name
      </label>
      <input
        aria-label="Name your flight"
        className="search-box"
        id="FlightForm"
        name="name"
        placeholder="Input flight name"
        required={true}
        type="text"
      />
      <label
        htmlFor="DepartureAirport"
        style={
          Object {
            "textAlign": "left",
          }
        }
      >
        Departure Airport
      </label>
      <input
        aria-label="Departure airport"
        className="search-box"
        id="FlightForm"
        name="departure_airport"
        placeholder="Input departure airport"
        required={true}
        type="text"
      />
      <label
        htmlFor="ArrivalAirport"
        style={
          Object {
            "textAlign": "left",
          }
        }
      >
        Arrival Airport
      </label>
      <input
        aria-label="Arrival airport"
        className="search-box"
        id="FlightForm"
        name="arrival_airport"
        placeholder="Input arrival airport"
        required={true}
        type="text"
      />
      <label
        htmlFor="Airline"
        style={
          Object {
            "textAlign": "left",
          }
        }
      >
        Airline
      </label>
      <input
        aria-label="Airline company"
        className="search-box"
        id="FlightForm"
        name="company"
        placeholder="Input airline company"
        required={true}
        type="text"
      />
      <label
        htmlFor="ArrivalTime"
        style={
          Object {
            "textAlign": "left",
          }
        }
      >
        Arrival Time
      </label>
      <input
        aria-label="Arrival time"
        className="search-box"
        id="FlightForm"
        name="arrival_time"
        placeholder="Input arrival time"
        required={true}
        type="time"
      />
      <label
        htmlFor="DepartureTime"
        style={
          Object {
            "textAlign": "left",
          }
        }
      >
        Departure Time
      </label>
      <input
        aria-label="Departure time"
        className="search-box"
        id="FlightForm"
        name="departure_time"
        placeholder="Input departure time"
        required={true}
        type="time"
      />
      <button
        aria-label="Create flight"
      >
        Create Flight
      </button>
      <button
        aria-label="Reset"
        className="reset"
        type="reset"
      >
        Reset
      </button>
      <button
        aria-label="Cancel"
        className="cancel"
        onClick={[Function]}
        type="cancel"
      >
        Cancel
      </button>
    </section>
  </form>
</div>
`;

exports[`FlightForm component renders the FlightForm given props 1`] = `
<div
  className="AddFlight"
>
  <Navbar />
  <form
    className="add-flight"
    onSubmit={[Function]}
  >
    <section
      className="aircraft"
    >
      <h1
        style={
          Object {
            "fontSize": "30px",
            "textAlign": "center",
          }
        }
      >
        Create your flight
      </h1>
      <label
        htmlFor="FlightName"
        style={
          Object {
            "textAlign": "left",
          }
        }
      >
        Flight Name
      </label>
      <input
        aria-label="Name your flight"
        className="search-box"
        id="FlightForm"
        name="name"
        placeholder="Input flight name"
        required={true}
        type="text"
      />
      <label
        htmlFor="DepartureAirport"
        style={
          Object {
            "textAlign": "left",
          }
        }
      >
        Departure Airport
      </label>
      <input
        aria-label="Departure airport"
        className="search-box"
        id="FlightForm"
        name="departure_airport"
        placeholder="Input departure airport"
        required={true}
        type="text"
      />
      <label
        htmlFor="ArrivalAirport"
        style={
          Object {
            "textAlign": "left",
          }
        }
      >
        Arrival Airport
      </label>
      <input
        aria-label="Arrival airport"
        className="search-box"
        id="FlightForm"
        name="arrival_airport"
        placeholder="Input arrival airport"
        required={true}
        type="text"
      />
      <label
        htmlFor="Airline"
        style={
          Object {
            "textAlign": "left",
          }
        }
      >
        Airline
      </label>
      <input
        aria-label="Airline company"
        className="search-box"
        id="FlightForm"
        name="company"
        placeholder="Input airline company"
        required={true}
        type="text"
      />
      <label
        htmlFor="ArrivalTime"
        style={
          Object {
            "textAlign": "left",
          }
        }
      >
        Arrival Time
      </label>
      <input
        aria-label="Arrival time"
        className="search-box"
        id="FlightForm"
        name="arrival_time"
        placeholder="Input arrival time"
        required={true}
        type="time"
      />
      <label
        htmlFor="DepartureTime"
        style={
          Object {
            "textAlign": "left",
          }
        }
      >
        Departure Time
      </label>
      <input
        aria-label="Departure time"
        className="search-box"
        id="FlightForm"
        name="departure_time"
        placeholder="Input departure time"
        required={true}
        type="time"
      />
      <button
        aria-label="Create flight"
      >
        Create Flight
      </button>
      <button
        aria-label="Reset"
        className="reset"
        type="reset"
      >
        Reset
      </button>
      <button
        aria-label="Cancel"
        className="cancel"
        onClick={[Function]}
        type="cancel"
      >
        Cancel
      </button>
    </section>
  </form>
</div>
`;
